lib/modules/manager/cargo/update.ts

```diff
@@ -0,0 +1,56 @@
+import semver, { ReleaseType } from 'semver';
+import { logger } from '../../../logger';
+import { regEx } from '../../../util/regex';
+import type { BumpPackageVersionResult } from '../types';
+
+export function bumpPackageVersion(
+  content: string,
+  currentValue: string,
+  bumpVersion: ReleaseType,
+): BumpPackageVersionResult {
+  logger.debug(
+    { bumpVersion, currentValue },
+    'Checking if we should bump Cargo.toml version',
+  );
+  let bumpedContent = content;
+
+  if (!semver.valid(currentValue)) {
+    logger.warn(
+      { currentValue },
+      'Unable to bump Cargo.toml version, not a valid semver',
+    );
+    return { bumpedContent };
+  }
+
+  try {
+    const newCrateVersion = semver.inc(currentValue, bumpVersion);
+    if (!newCrateVersion) {
+      throw new Error('semver inc failed');
+    }
+
+    logger.debug({ newCrateVersion });
+
+    bumpedContent = content.replace(
```

Comment:
Isn't it more reliable to replace current value with new value? Or is there no chance of mismatches?

Reply:
I am not sure what you mean. Could you elaborate?

Reply:
Ah, I see what you mean. Indeed, there is a way to define a dependency like so:

```
[dependencies.awesome]
version = "1.3.5"
default-features = false
features = ["secure-password", "civet"]
```

Which would cause a conflict. I am not familiar with the codebase, so I am not sure what the most straightforward way to fix this is.

Could you elaborate on this:
> Isn't it more reliable to replace current value with new value?

Reply:
I should point out that I modeled this PR on #26361 (also used in #26377) which is doing the same thing with the regex, but supports this format as well (take from [here](https://python-poetry.org/docs/dependency-specification/#expanded-dependency-specification-syntax)):

```
[tool.poetry.group.dev.dependencies.black]
version = "19.10b0"
allow-prereleases = true
python = "^3.7"
markers = "platform_python_implementation == 'CPython'"
```

This would mean these have a bug as well, correct?

Reply:
What I meant was that your replace function "from" part should include the extracted currentValue to reduce the chances of mistaken replace.

I wouldn't describe it as a bug but it's possible it could have a mistaken replacement

Reply:
Actually, I think this is OK for now. If someone can reproduce a problem later, we can enhance the logic then.